I also saw this error:

Using the new science lut:
imap_codice_l1a-sci-lut_20251218_v001.json

I am getting an error on line 224 in codice_l1a_lo_species.py:
l1a_dataset["acquisition_time_per_step"] = xr.DataArray(
    calculate_acq_time_per_step(sci_lut_data["lo_stepping_tab"]),
    dims=("esa_step",),
    attrs=cdf_attrs.get_variable_attributes(
        "acquisition_time_per_step", check_schema=False
    ),
)

E               xarray.structure.alignment.AlignmentError: cannot reindex or align along dimension 'esa_step' because of conflicting dimension sizes: {104, 128} (note: an index is found along that dimension with size=128)

venv/lib/python3.12/site-packages/xarray/structure/alignment.py:451: AlignmentError

But maybe this is for another PR.

--- a/imap_processing/codice/codice_l2.py
+++ b/imap_processing/codice/codice_l2.py
@@ -42,6 +42,7 @@ from imap_processing.codice.constants import (
     SW_POSITIONS,
 )
 from imap_processing.codice.utils import apply_replacements_to_attrs
+from imap_processing.spice.time import et_to_datetime64, ttj2000ns_to_et
 
 logger = logging.getLogger(__name__)
 
@@ -320,11 +321,10 @@ def compute_geometric_factors(
     #  2026. We also need to fix this on days when the sci Lut changes.
     # After November 24th 2025 we need to do this step a different way.
     date_switch = datetime.datetime(2025, 11, 24)
-    start_date = dataset.attrs.get("Logical_file_id", None)
-    if start_date is None:
-        raise ValueError("Dataset is missing Logical_file_id attribute.")
-    processing_date = datetime.datetime.strptime(start_date.split("_")[4], "%Y%m%d")
-    if processing_date < date_switch:
+    dataset_midpoint_ns = dataset["epoch"].values[dataset["epoch"].size // 2]
+    # Convert to datetime64 for comparison
+    dataset_midpoint = et_to_datetime64(ttj2000ns_to_et(dataset_midpoint_ns))
+    if dataset_midpoint < date_switch:
         modes = (half_spin_per_esa_step > rgfo_half_spin) & (rgfo_half_spin > 0)
     else:
         # After November 24th, 2025, we no longer apply reduced geometric factors;

--- a/imap_processing/tests/codice/test_codice_l2.py
+++ b/imap_processing/tests/codice/test_codice_l2.py
@@ -113,6 +113,36 @@ def test_compute_geometric_factors_all_full_mode(mock_half_spin_per_esa_step):
     np.testing.assert_array_equal(result, expected)
 
 
+def test_compute_geometric_factors_past_nov_24th(mock_half_spin_per_esa_step):
+    # rgfo_half_spin = 1 means all half_spin values (>=2) are >= rgfo_half_spin
+    # Although the rgfo_half_spin indicates reduced mode, the date is past Nov 24th,
+    # 2024 so we expect full mode to be used.
+    dataset = xr.Dataset(
+        {
+            "rgfo_half_spin": (("epoch",), np.array([1, 1])),
+            "half_spin_per_esa_step": (
+                (
+                    "epoch",
+                    "esa_step",
+                ),
+                np.tile(mock_half_spin_per_esa_step, (2, 1)),
+            ),
+        },
+    )
+    geometric_factor_lut = {
+        "full": np.zeros((128, 24)),
+        "reduced": np.ones((128, 24)),
+    }
+    # Make sure epoch is past Nov 24th, 2024
+    ns_past_nov_24 = 900000000000000000
+    dataset = dataset.assign_coords({"epoch": np.repeat(ns_past_nov_24, 2)})
+    result = compute_geometric_factors(dataset, geometric_factor_lut)
+
+    # Expect "full" values everywhere
+    expected = np.full((2, 128, 24), 0)
+    np.testing.assert_array_equal(result, expected)
+
+
 def test_compute_geometric_factors_all_reduced_mode(mock_half_spin_per_esa_step):
     # rgfo_half_spin = 1 means all half_spin values (>=2) are >= rgfo_half_spin
     dataset = xr.Dataset(

--- a/imap_processing/tests/codice/test_codice_l2.py
+++ b/imap_processing/tests/codice/test_codice_l2.py
@@ -101,7 +101,6 @@ def test_compute_geometric_factors_all_full_mode(mock_half_spin_per_esa_step):
             "rgfo_half_spin": (("epoch",), np.array([4, 4])),
             "half_spin_per_esa_step": (("esa_step",), mock_half_spin_per_esa_step),
         },
-        attrs={"Logical_file_id": "imap_codice_l1b_lo-sw-angular_20250101_v001"},
     )
     geometric_factor_lut = {
         "full": np.zeros((128, 24)),
@@ -121,7 +120,6 @@ def test_compute_geometric_factors_all_reduced_mode(mock_half_spin_per_esa_step)
             "rgfo_half_spin": (("epoch",), np.array([1])),
             "half_spin_per_esa_step": (("esa_step",), mock_half_spin_per_esa_step),
         },
-        attrs={"Logical_file_id": "imap_codice_l1b_lo-sw-angular_20250101_v001"},
     )
     geometric_factor_lut = {
         "full": np.zeros((128, 24)),
@@ -141,7 +139,6 @@ def test_compute_geometric_factors_mixed(mock_half_spin_per_esa_step):
             "rgfo_half_spin": (("epoch",), np.array([2])),
             "half_spin_per_esa_step": (("esa_step",), mock_half_spin_per_esa_step),
         },
-        attrs={"Logical_file_id": "imap_codice_l1b_lo-sw-angular_20250101_v001"},
     )
     geometric_factor_lut = {
         "full": np.zeros((128, 24)),

